### ISSUE TITLE
Fix S3433 FP: Rule should not report on internal MSTest methods when DiscoverInternals is enabled

### ISSUE THREAD

**Issue by HugoLemieux** [SonarSource/sonar-dotnet#9844]
### Description

Rule S3433 reports a code smell on MSTest methods declared as internal:

Make this test method 'public'.

However, this is a false positive when the test assembly enables MSTest's DiscoverInternalsAttribute, which allows MSTest to discover and execute internal test classes and test methods. Microsoft documentation explicitly states that internal test methods are supported in this scenario.

### Reproducer

DiscoverInternals.cs
```
using Microsoft.VisualStudio.TestTools.UnitTesting;

[assembly: DiscoverInternals]
```
Test class
```
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace MyProject.Tests;

[TestClass]
internal class SampleTests
{
    [TestMethod]
    internal void ShouldPass()
    {
        Assert.IsTrue(true);
    }
}
```

### Product and Version

SonarQube Server v26.1.0.118079
